src/_pytest/pathlib.py

```diff
@@ -59,13 +59,14 @@ def on_rm_rf_error(func, path: str, exc, *, start_path) -> bool:
         return False
 
     if func not in (os.rmdir, os.remove, os.unlink):
-        warnings.warn(
-            PytestWarning(
-                "(rm_rf) unknown function {} when removing {}:\n{}: {}".format(
-                    path, func, exctype, excvalue
+        if func not in (os.open,):
```

Comment:
shouldn't we look at retrying for a directory there?

Reply:
IIRC it is not possible to rety just that call.  (see the other comments about this)

Reply:
based on https://github.com/python/cpython/blob/a8fb9327fb9387b404d4b6ce55c608125f66b9ae/Lib/shutil.py#L637-L642

we cant retry, this is frickle

Reply:
My idea initially was to retry the whole operation, not just the function call (IIRC there's an older/closed PR with that attempt).
Currently we would also emit a warning for e.g. `os.lstat` - a warning for "unknown function" in that context also sounds strange.
As of now I think we should have no warnings for unhandled functions, or whitelist all the ones being used.

Reply:
thanks for the research